libs/core/langchain_core/documents/base.py

```diff
@@ -1,47 +1,269 @@
 from __future__ import annotations
 
-from typing import Any, List, Literal, Optional
+import contextlib
+import mimetypes
+from io import BufferedReader, BytesIO
+from pathlib import PurePath
+from typing import Any, Generator, List, Literal, Mapping, Optional, Union, cast
 
 from langchain_core.load.serializable import Serializable
-from langchain_core.pydantic_v1 import Field
+from langchain_core.pydantic_v1 import Field, root_validator
 
+PathLike = Union[str, PurePath]
 
-class Document(Serializable):
-    """Class for storing a piece of text and associated metadata.
-
-    Example:
 
-        .. code-block:: python
+class BaseContent(Serializable):
+    """Base class to represent content.
 
-            from langchain_core.documents import Document
+    LangChain content objects are used to represent raw data, such as text or
+    binary data.
 
-            document = Document(
-                page_content="Hello, world!",
-                metadata={"source": "https://example.com"}
-            )
+    All such objects are expected to allow specifying metadata and an
+    optional identifier.
     """
 
     # The ID field is optional at the moment.
     # It will likely become required in a future major release after
     # it has been adopted by enough vectorstore implementations.
     id: Optional[str] = None
     """An optional identifier for the document.
-    
+
     Ideally this should be unique across the document collection and formatted 
     as a UUID, but this will not be enforced.
     """
 
-    page_content: str
-    """String text."""
     metadata: dict = Field(default_factory=dict)
-    """Arbitrary metadata about the page content (e.g., source, relationships to other
-        documents, etc.).
+    """Arbitrary metadata associated with the content."""
+
+
+class Blob(BaseContent):
```

Comment:
Probably slightly too late for this but could see `Blob` being a confusing name given it shadows the actual MDN concept name

Reply:
It already exists on the python side, but we can introduce an alias. I wasn't thinking about name collisions on the JS side